packages/react-api/src/hooks/useCartoLayerProps.js

```diff
@@ -1,19 +1,29 @@
 import { DataFilterExtension } from '@deck.gl/extensions';
 import { _buildFeatureFilter } from '@carto/react-core';
-import useViewportFeatures from './useViewportFeatures';
 import { MAP_TYPES, API_VERSIONS } from '@deck.gl/carto';
+import { useSelector } from 'react-redux';
+import useGeoJsonFeatures from './useGeoJsonFeatures';
+import useTilesetFeatures from './useTilesetFeatures';
 
 export default function useCartoLayerProps({
   source,
   uniqueIdProperty,
   viewportFeatures = true,
-  viewporFeaturesDebounceTimeout = 500
+  viewporFeaturesDebounceTimeout = 250
 }) {
-  const [onViewportLoad, onDataLoad, fetch] = useViewportFeatures(
-    source,
+  const viewport = useSelector((state) => state.carto.viewport);
+
+  const [onDataLoad] = useGeoJsonFeatures(source, {
+    viewport,
+    uniqueIdProperty,
+    debounceTimeout: viewporFeaturesDebounceTimeout
+  });
```

Comment:
I wouldn't change to positional params with 'source' here, like we discussed in the RFC, for consistency reasons. I would keep the 'set of params', with:

```
  const [onDataLoad] = useGeoJsonFeatures({
    source, 
    uniqueIdProperty,
    debounceTimeout: viewporFeaturesDebounceTimeout
  })
```

And if you really want to create 2 levels, with optional params:

```
  const [onDataLoad] = useGeoJsonFeatures({
    source, 
    options: { 
           uniqueIdProperty,
           debounceTimeout: viewporFeaturesDebounceTimeout
     }
  })
```

Reply:
Updated.